src/ipc/ipc4/handler.c

```diff
@@ -1494,8 +1494,11 @@ struct ipc_cmd_hdr *ipc_prepare_to_send(const struct ipc_msg *msg)
 	msg_data.msg_out.pri = msg->header;
 	msg_data.msg_out.ext = msg->extension;
 
-	if (msg->tx_size)
+	if (msg->tx_size) {
```

Comment:
it should be done only if the message is coming from secondary core, otherwise you may loose data that are stored in primary core cache. 
Add .shared field to the structure and do invalidate only if the message is shared cross-core